### PR TITLE
fixed issue #174 removed runtime checking on Projectile and moved git…

### PR DESCRIPTION
**Pull request by edewit**
… operations to GitHubService

Conflicts:
	core/core-api/src/main/java/org/kontinuity/catapult/core/api/CreateProjectileBuilder.java
	core/core-api/src/main/java/org/kontinuity/catapult/core/api/ForkProjectileBuilder.java
	core/core-api/src/main/java/org/kontinuity/catapult/core/api/Projectile.java
	core/core-impl/pom.xml
	core/core-impl/src/main/java/org/kontinuity/catapult/core/impl/CatapultImpl.java
	core/core-impl/src/test/java/org/kontinuity/catapult/core/impl/CatapultIT.java
	services/github-service-impl/pom.xml
	services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/impl/kohsuke/GitHubServiceFactoryImpl.java
	services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/impl/kohsuke/KohsukeGitHubServiceImpl.java
	web/src/main/java/org/kontinuity/catapult/web/api/CatapultResource.java

- [X] Have you created an [issue](https://github.com/redhat-kontinuity/catapult/issues) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/redhat-kontinuity/catapult/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

-----

use this one in favor of #176

### CODE REVIEW

File: services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/impl/kohsuke/KohsukeGitHubServiceImpl.java [hunk 1]
```diff
@@ -166,6 +176,24 @@ public GitHubRepository createRepository(String repositoryName,
         return wrapped;
     }
 
+    @Override
+    public GitHubRepository createRepository(String repositoryName, String description, File path) throws IllegalArgumentException {
```
Comment (gastaldi) [inline services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/impl/kohsuke/KohsukeGitHubServiceImpl.java]: I think we should rename this method signature to something like `push(File path, GithubRepository)`.

File: core/core-impl/src/main/java/org/kontinuity/catapult/core/impl/CatapultImpl.java [hunk 2]
```diff
@@ -50,91 +43,86 @@
...
+                .path(forkProjectile.getGitRef())
+                .path(forkProjectile.getPipelineTemplatePath()).build();
+
+        // Configure the OpenShift project
+        openShiftService.configureProject(createdProject,
+                                          gitHubRepository.getGitCloneUri(),
+                                          forkProjectile.getGitRef(),
+                                          pipelineTemplateUri);
+
+        GitHubWebhook webhook = getGitHubWebhook(gitHubService, gitHubRepository, createdProject);
```
Comment (gastaldi) [inline core/core-impl/src/main/java/org/kontinuity/catapult/core/impl/CatapultImpl.java]: getWebhook Creates webhook, maybe we should revisit this method name

File: web/src/main/java/org/kontinuity/catapult/web/api/CatapultResource.java [hunk 3]
```diff
@@ -221,34 +208,11 @@ private Response fling(Projectile projectile) {
...
-        }
-
-        Projectile getProjectile() {
-            return projectile;
-        }
-
-        Response getResponse() {
-            return response;
-        }
+    private CreateProjectile createCreateProjectile(String path, String gitHubAccessToken) {
```
Comment (gastaldi) [inline web/src/main/java/org/kontinuity/catapult/web/api/CatapultResource.java]: Isn't that what the builder is for? Why another method?

Comment (edewit) [inline web/src/main/java/org/kontinuity/catapult/web/api/CatapultResource.java]: just so that I don't have to duplicate this code twice
```java
ProjectileBuilder.newInstance()
                .gitHubAccessToken(gitHubAccessToken)
                .createType()
                .projectLocation(path)
                .build();
```

File: core/core-impl/src/main/java/org/kontinuity/catapult/core/impl/CatapultImpl.java (hunk 2 above)
Comment (edewit) [inline core/core-impl/src/main/java/org/kontinuity/catapult/core/impl/CatapultImpl.java]: this is not code that I introduced, we can update it, but let's do that in a separate PR

File: web/src/main/java/org/kontinuity/catapult/web/api/CatapultResource.java (hunk 3 above)
Comment (gastaldi) [inline web/src/main/java/org/kontinuity/catapult/web/api/CatapultResource.java]: But isn't that the same as calling this method twice? I thought the Builder would be a replacement for factory methods
